File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/dashboard/DashboardContract.java
package com.tai.androidtai.modules.dashboard;


import com.tai.androidtai.domain.bean.MealBean;
import com.tai.androidtai.modules.core.BaseContract;

import java.util.List;

public interface DashboardContract {

    interface View extends BaseContract.View {

        void displayInformation(List<MealBean> userList);
    }

    interface Presenter extends BaseContract.Presenter {

        void getInfo();

        void goToMealDetails(MealBean mealBean);
    }

    interface Router extends BaseContract.Router {
        void goToMealDetails(MealBean mealBean);
    }

}

File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/dashboard/DashboardContract.kt
package com.tai.androidtai.modules.dashboard

import com.tai.androidtai.domain.bean.ResultBean
import com.tai.androidtai.modules.core.BaseContract

interface DashboardContract {

    interface View : BaseContract.View {

        fun displayInformation(userList: List<ResultBean>?)

        fun hideProgressBar()
    }

    interface Presenter : BaseContract.Presenter {

        fun getInfo()

        fun goToMealDetails(resultBean: ResultBean, price: String)
    }

    interface Router : BaseContract.Router {
        fun goToMealDetails(resultBean: ResultBean, price: String, view: View?)
    }

}
